server/src/test/java/org/elasticsearch/search/SearchUtilsTests.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.search;
+
+import org.elasticsearch.test.ESTestCase;
+
+public class SearchUtilsTests extends ESTestCase {
+
+    public void testConfigureMaxClauses() {
+
+        // Heap below 1 Gb
+        assertEquals(4096, SearchUtils.calculateMaxClauseValue(4, 0));
+
+        // Number of processors not available
+        assertEquals(4096, SearchUtils.calculateMaxClauseValue(-1, 1));
```

Comment:
is it actually something that can happen? While the number of cores can be unknown, we are actually using the threadpool size, which should always be defined?

Reply:
I'm fairly sure that this would always be set, but put this in as a safety net.  But maybe it should be replaced with an assertion that the thread pool size > 1